Framework/PythonInterface/mantid/simpleapi.py

```diff
@@ -1407,3 +1407,49 @@ def _attach_algorithm_func_as_method(method_name, algorithm_wrapper, algm_object
                                              algm_object.workspaceMethodOn())
 
 # -------------------------------------------------------------------------------------------------------------
+
+
+def write_all_workspaces_histories(folder):
```

Comment:
Can you help me understand why we wrote the code in Python and not in C++? At this level these functions will be available to all users automatically - was that the intention?

I can't quite see why we have some code here and then code in the `scripts/Recovery` directory.

Reply:
Originally this was based on spike work written in Python

As we now only save out WS histories with the timestamp concatenated I agree we move this element to C++.

In a future PR I've converted `OrderWorkspaceHistories` into a utility algorithm which we can execute directly too.

Reply:
:+1: